tools/cassandra_delete_range/main.go

```diff
@@ -0,0 +1,502 @@
+	return firstLedgerIdx, latestLedgerIdx, nil
+}
+
+func deleteLedgerData(cluster *gocql.ClusterConfig, fromLedgerIdx uint64, toLedgerIdx uint64) error {
+	log.Printf("Start generating delete queries for %d -> %d\n\n", fromLedgerIdx, toLedgerIdx)
+
+	var traversedRowsCount uint64
+	var info deleteInfo
+
+	// successor queries
+	log.Println("Generating delete queries for successor table")
+	info, traversedRowsCount = prepareDeleteQueries(cluster,
+		fromLedgerIdx, toLedgerIdx,
+		"SELECT key, seq FROM successor WHERE token(key) >= %s AND token(key) <= %s",
+		"DELETE FROM successor WHERE key = ? AND seq = ?")
+	log.Printf("Total delete queries: %d\n", len(info.Data))
+	log.Printf("Total traversed rows: %d\n\n", traversedRowsCount)
+	performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: true})
+
+	// diff queries
+	log.Println("Generating delete queries for diff table")
+	info, traversedRowsCount = prepareDeleteQueries(cluster,
+		fromLedgerIdx, toLedgerIdx,
+		"SELECT key, seq FROM diff WHERE token(seq) >= %s AND token(seq) <= %s",
+		"DELETE FROM diff WHERE key = ? AND seq = ?")
+	log.Printf("Total delete queries: %d\n", len(info.Data))
+	log.Printf("Total traversed rows: %d\n\n", traversedRowsCount)
+	performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: true})
+
+	// objects queries
+	log.Println("Generating delete queries for objects table")
+	info, traversedRowsCount = prepareDeleteQueries(cluster,
+		fromLedgerIdx, toLedgerIdx,
+		"SELECT key, sequence FROM objects WHERE token(key) >= %s AND token(key) <= %s",
+		"DELETE FROM objects WHERE key = ? AND sequence = ?")
+	log.Printf("Total delete queries: %d\n", len(info.Data))
+	log.Printf("Total traversed rows: %d\n\n", traversedRowsCount)
+	performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: true})
+
+	// ledger_hashes queries
+	log.Println("Generating delete queries for ledger_hashes table")
+	info, traversedRowsCount = prepareDeleteQueries(cluster,
+		fromLedgerIdx, toLedgerIdx,
+		"SELECT hash, sequence FROM ledger_hashes WHERE token(hash) >= %s AND token(hash) <= %s",
+		"DELETE FROM ledger_hashes WHERE hash = ?")
+	log.Printf("Total delete queries: %d\n", len(info.Data))
+	log.Printf("Total traversed rows: %d\n\n", traversedRowsCount)
+	performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: false})
+
+	// transactions queries
+	log.Println("Generating delete queries for transactions table")
+	info, traversedRowsCount = prepareDeleteQueries(cluster,
+		fromLedgerIdx, toLedgerIdx,
+		"SELECT hash, ledger_sequence FROM transactions WHERE token(hash) >= %s AND token(hash) <= %s",
+		"DELETE FROM transactions WHERE hash = ?")
+	log.Printf("Total delete queries: %d\n", len(info.Data))
+	log.Printf("Total traversed rows: %d\n\n", traversedRowsCount)
+	performDeleteQueries(cluster, &info, columnSettings{UseBlob: true, UseSeq: false})
+
+	// ledger_transactions queries
```

Comment:
ledger_transactions partition key seems be ledger_sequence , maybe no need whole table scan.